### ISSUE TITLE
'develop' mode build error in node 16

### ISSUE THREAD

**Issue by dannon** [galaxyproject/galaxy-hub#1207]
When entering malformed markdown(needs a blank line between html and md contents), like:

```
<p>
Sample markdown block with link [here][link].
</p>
```

In node 14, you get a friendly error that scrolls by too fast to see:
```
(node:22254) UnhandledPromiseRejectionWarning: Error: Could not find node to update with id: fc12f98370e4b3dd2ffa72a758d4cf7c
    at Collection.updateNode (/home/dannon/work/galaxy-hub/node_modules/gridsome/lib/store/Collection.js:124:13)
    at FSWatcher.<anonymous> (/home/dannon/work/galaxy-hub/node_modules/@gridsome/source-filesystem/index.js:117:36)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:22254) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 41)
(node:22254) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:22254) UnhandledPromiseRejectionWarning: Error: Could not find node to update with id: fc12f98370e4b3dd2ffa72a758d4cf7c
    at Collection.updateNode (/home/dannon/work/galaxy-hub/node_modules/gridsome/lib/store/Collection.js:124:13)
    at FSWatcher.<anonymous> (/home/dannon/work/galaxy-hub/node_modules/@gridsome/source-filesystem/index.js:117:36)
(node:22254) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 42)
```

But in node 16, the node process is killed, requiring a restart.

This isn't in our code, so the best fix would be for gridsome to fix this upstream.  Barring that, we could handle this more gracefully in two potential ways.  1) automatically restart the gridsome develop subprocess, and maybe 2) show an informative message to the user saying why they're seeing it (bad markdown entered), and suggesting using node 14 for now.

**Comment by dannon** [comment]
To update here, we've been interactively tracking this down and the invalid markdown was a red herring.  The error still happens because it looks like gridsome's collection code is deleting the node from the collection prior to the update.  Debugging continues and I'll update here with more.

**Comment by dannon** [comment]
Fixed with #1213, though we might want to follow up and make this more robust somehow.

Copying over some debugging output explaining the problem:

Looks like the .webp was a symlink on the initial build, if there's some logic you want to adjust there.  If it's a symlink, is there no node and it's simply copied as static, which would explain this I guess?

```
9b95a70f52248b68d25bbec3451accc0 00ebd1ae2e91195e9f7e05654201f218 /build/content-md/usegalaxy/welcome/main/
fc12f98370e4b3dd2ffa72a758d4cf7c 83d8dce80b28555e13c6f5d16ac6bb05 /build/content-md/usegalaxy/welcome/jxtx-foundation/
/home/dannon/work/galaxy-hub/node_modules/gridsome/lib/store/Collection.js:132
      throw new Error(`STILL BUSTED Could not find node to update with id: ${options.id}`)
            ^

Error: STILL BUSTED Could not find node to update with id: fc12f98370e4b3dd2ffa72a758d4cf7c
    at Collection.updateNode (/home/dannon/work/galaxy-hub/node_modules/gridsome/lib/store/Collection.js:132:13)
    at FSWatcher.<anonymous> (/home/dannon/work/galaxy-hub/node_modules/@gridsome/source-filesystem/index.js:117:36)
```

And that output comes from dumping `id`, `$uid`, and `path` in `updateNode` in the Collection module of gridsome seen in the stacktrace.

**Comment by NickSto** [comment]
Alright, it turns out that, as I suspected, this is very related to similar errors I've seen before.

#748 is not exactly this situation, but the core issue is also symlinks. And it turns out the solution also prevents this crash (evidence below). The solution basically tells `partition-content.mjs` to always place full copies of all content files into the build directories (never symlinks).

The drawback, and the reason the solution isn't the default behavior, is that this means the build directories take far more space, since the entire content is duplicated. In my test, the build directories take 337 MB when using copies, instead of the 37MB when using symlinks. Interestingly partitioning doesn't take any more time, though.

I suppose we should go ahead and make this behavior the default. It'd solve this and also put #748 to bed. It'll just cost another couple hundred MB for our authors.

## Test case

I installed Node 17, checked out ecaeb0f7d, ran `yarn install`, and then ran the development server:
1. First, I ran plain `yarn develop`, then after it booted, I made a small text edit to `content/usegalaxy/welcome/main.md` and it crashed.
2. Then, I ran `yarn develop --resource copy`, then did the same thing. It worked fine, not even a warning.

**Comment by NickSto** [comment]
Fixed by #1214.
